pandas/io/excel.py

```diff
@@ -465,7 +465,7 @@ def _parse_cell(cell_contents, cell_typ):
                 if not squeeze or isinstance(output[asheetname], DataFrame):
                     output[asheetname].columns = output[
                         asheetname].columns.set_names(header_names)
-            except StopIteration:
+            except EmptyDataError:
```

Comment:
there are 2 cases here, a ``ValueError`` and an ``EmptyDataError``. you should catch the ``EmptyDataError`` and continue, but NOT a plain ``ValueError``